app/controllers/auth/registrations_controller.rb

```diff
@@ -76,4 +95,18 @@ def password_update?
     !user[:current_password].blank? || !user[:password].blank? ||
       !user[:password_confirmation].blank?
   end
+
+  # Returns whether the given user can be disabled or not. The following rules
+  # apply:
+  #   1. A user can disable himself unless it's the last admin on the system.
+  #   2. The admin user is the only one that can disable other users.
+  def can_disable?(user)
+    if current_user == user
+      # An admin cannot disable himself if he's the only admin in the system.
+      current_user.admin? && User.admins.count == 1
+    else
+      # Only admin users can disable other users.
+      !current_user.admin?
```

Comment:
This part of the branch is useless because we have the `check_admin` filter in place. So a non admin user is never going to reach this part of the code